modules/apps/commerce/commerce-payment-method-money-order/src/main/java/com/liferay/commerce/payment/method/money/order/internal/portlet/action/EditMoneyOrderCommercePaymentMethodConfigurationMVCActionCommand.java

```diff
@@ -80,13 +82,26 @@ private void _updateCommercePaymentMethod(ActionRequest actionRequest)
 		ModifiableSettings modifiableSettings =
 			settings.getModifiableSettings();
 
+		String showMessagePage = ParamUtil.getString(
+			actionRequest, "settings--showMessagePage--");
+
+		modifiableSettings.setValue("showMessagePage", showMessagePage);
+
 		UnicodeProperties unicodeProperties = PropertiesParamUtil.getProperties(
-			actionRequest, "settings--");
+			actionRequest, "settings--messageAsLocalizedXML_");
+
+		Map<String, String> messageMap = new HashMap<>();
 
 		for (Map.Entry<String, String> entry : unicodeProperties.entrySet()) {
-			modifiableSettings.setValue(entry.getKey(), entry.getValue());
+			messageMap.put(entry.getKey(), entry.getValue());
 		}
 
+		String messageAsLocalizedXML = LocalizationUtil.getXml(
+			messageMap, "", "messageAsLocalizedXML");
```

Comment:
@luca-pellizzon we can use `StringPool.BLANK` here.
```suggestion
			messageMap, StringPool.BLANK, "messageAsLocalizedXML");
```